ledger/participant-integration-api/src/main/scala/platform/store/dao/events/EventsTableFlatEvents.scala

```diff
@@ -43,30 +44,53 @@ private[events] object EventsTableFlatEvents {
-          transactionId = transactionId,
-          nodeIndex = nodeIndex,
-          eventSequentialId = eventSequentialId,
-          ledgerEffectiveTime = ledgerEffectiveTime,
-          commandId = commandId.getOrElse(""),
-          workflowId = workflowId.getOrElse(""),
-          event = Raw.FlatEvent.Archived(
-            eventId = eventId,
-            contractId = contractId,
-            templateId = templateId,
-            eventWitnesses = ArraySeq.unsafeWrapArray(eventWitnesses),
-          ),
-        )
+  private val exercisedFlatEventParser: RowParser[EventsTable.Entry[Raw.FlatEvent]] =
+    EventsTable.exercisedEventRow.map {
+      case eventOffset ~ transactionId ~ nodeIndex ~ eventSequentialId ~ eventId ~ contractId ~ ledgerEffectiveTime ~ templateId ~ commandId ~ workflowId ~ eventWitnesses ~ exerciseConsuming ~ exerciseChoice ~ exerciseArgument ~ exerciseArgumentCompression ~ exerciseResult ~ exerciseResultCompression ~ exerciseActors ~ exerciseChildEventIds =>
+        if (exerciseConsuming)
+          EventsTable.Entry(
+            eventOffset = eventOffset,
+            transactionId = transactionId,
+            nodeIndex = nodeIndex,
+            eventSequentialId = eventSequentialId,
+            ledgerEffectiveTime = ledgerEffectiveTime,
+            commandId = commandId.getOrElse(""),
+            workflowId = workflowId.getOrElse(""),
+            event = Raw.FlatEvent.Archived(
+              eventId = eventId,
+              contractId = contractId,
+              templateId = templateId,
+              eventWitnesses = ArraySeq.unsafeWrapArray(eventWitnesses),
+            ),
+          )
+        else
+          EventsTable.Entry(
+            eventOffset = eventOffset,
+            transactionId = transactionId,
+            nodeIndex = nodeIndex,
+            eventSequentialId = eventSequentialId,
+            ledgerEffectiveTime = ledgerEffectiveTime,
+            commandId = commandId.getOrElse(""),
+            workflowId = workflowId.getOrElse(""),
+            event = Raw.FlatEvent.Exercised(
+              eventId = eventId,
+              contractId = contractId,
+              templateId = templateId,
+              exerciseConsuming = exerciseConsuming,
+              exerciseChoice = exerciseChoice,
+              exerciseArgument = exerciseArgument,
+              exerciseArgumentCompression = exerciseArgumentCompression,
+              exerciseResult = exerciseResult,
+              exerciseResultCompression = exerciseResultCompression,
+              exerciseActors = ArraySeq.unsafeWrapArray(exerciseActors),
+              exerciseChildEventIds = ArraySeq.unsafeWrapArray(exerciseChildEventIds),
+              eventWitnesses = ArraySeq.unsafeWrapArray(eventWitnesses),
+            ),
+          )
     }
 
   val rawFlatEventParser: RowParser[EventsTable.Entry[Raw.FlatEvent]] =
-    createdFlatEventParser | archivedFlatEventParser
+    createdFlatEventParser | exercisedFlatEventParser
```

Comment:
Replace a simple parser for `Archived` events with a more generic parser for all exercised events. This is required because we may receive both consuming and non-consuming event rows now.